os/board/rtl8730e/src/component/soc/amebad2/fwlib/ram_common/ameba_i2c.c

```diff
@@ -701,9 +701,10 @@ void I2C_MasterReadDW(I2C_TypeDef *I2Cx, u8 *pBuf, u32 len)
 u32 I2C_MasterRead(I2C_TypeDef *I2Cx, u8 *pBuf, u32 len)
 {
 	u32 cnt = 0;
-
+	u32 timeout;
 	/* read in the DR register the data to be received */
 	for (cnt = 0; cnt < len; cnt++) {
+		timeout = 2000; /*timeout 2ms*/
```

Comment:
Is this proper value? If the len is much big, then, does not this value cause timeout at normal working?

Reply:
Yeah, because timeout will reset to 2000 (2ms) every time it enter the loop, so it is 2ms for each data, not for the entire data length.